lookout/style/format/analyzer.py

```diff
@@ -35,24 +40,83 @@ def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
     @with_uasts_and_contents
     def train(cls, ptr: ReferencePointer, config: dict, data_request_stub: DataStub,
               **data) -> AnalyzerModel:
-        cls.log.info("train %s %s %s", ptr.url, ptr.commit, data)
+        """
+        Train a model given the files available.
+
+        :param url: url of the repository
+        :param commit: hash of the commit that we analyze
+        :param config: configuration dict
+        :param data: data sent by the lookout server
+        :return: a modelforge.Model containing the learned rules, per language.
+        """
+        merged_config = {
+            "siblings_window": 5,
+            "parents_depth": 2,
+            "lower_bound_instances": 500,
+            "prune_branches_algorithms": ("reduced-error",),
+            "top_down_greedy_budget": TrainableRules.TopDownGreedyBudget(False, .5),
+            "prune_attributes": False,
+            "uncertain_attributes": True,
+            "prune_dataset_ratio": .2,
+            "n_estimators": 10,
+            "random_state": 42,
+            "n_iter": 30
+        }
+        merged_config.update(config)
+        cls.log.info("train %s %s %s with config %s", ptr.url, ptr.commit, data,
+                     str(merged_config))
```

Comment:
No need to `str()` the config. At the same time the order of keys will be random by default, so:

```python
from pprint import pformat
% pformat(final_config, width=4096, compact=True)
```